Assets/Shopify/UIToolkit/Themes/SingleProductTheme.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace Shopify.UIToolkit {
+    using Shopify.Unity;
+
+    /// <summary>
+    /// Override this class to create a theme that sells a single product
+    /// </summary>
+    public abstract class SingleProductTheme : ThemeBase {
+        /// <summary>
+        /// Called when the theme should show information about the product it's selling
+        /// </summary>
+        /// <param name="product">The product that should be displayed</param>
+        public abstract void OnShouldShowProduct(Product product);
```

Comment:
I thought about maybe also providing ProductVariant[] in this callback, to avoid implementors having to know how to dig through the gql structure of edges and nodes to get at the variants